crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -55,6 +58,41 @@ pub fn derive_reflect(input: TokenStream) -> TokenStream {
     }
 }
 
+#[proc_macro_derive(TypeName, attributes(module, type_name))]
+pub fn derive_type_name(input: TokenStream) -> TokenStream {
```

Comment:
Could you please add some documentation here that includes a basic summary of what this does as well as an example of how to use it (including specifying a custom type name)?